lightningd/memdump.c

```diff
@@ -154,6 +154,7 @@ static void scan_mem(struct command *cmd,
 	memleak_remove_htable(memtable, &ld->topology->txowatches.raw);
 	memleak_remove_htable(memtable, &ld->htlcs_in.raw);
 	memleak_remove_htable(memtable, &ld->htlcs_out.raw);
+	jsonrpc_remove_memleak(memtable, ld->jsonrpc);
```

Comment:
How is this not a leak?

Reply:
These functions are kind of misnamed :(

That just removes the pointers from the table of all pointers ("memtable"), so they don't count as leaks.

Reply:
But isn't `ld` supposed to be the root of allocations (and the only thing that is marked as not being a leak)? This is all reminiscent of mark and sweep garbage collection :wink: